### ISSUE TITLE
[feature request] Tab Organizer page

### ISSUE THREAD

**Issue by eternaleye** [piroor/treestyletab#1821]
Frequently, I'll end up opening a number of tabs in the course of browsing, which are located under whatever tab I discovered the link in (or at the bottom, in the case of control-T or opening links from pinned tabs). Often, these tabs belong to some category I already have a tab group I'd prefer to organize them under. These groups are at the top of the list, while new tabs are often at the end.

Currently, I have two options for organizing such tabs:

1. I can drag them, one-by-one, to the relevant groups, which is quite slow due to the need to wait for scrolling (and causes unloaded tabs to become loaded).
2. I can drag them to a new window, then drag from the new window to the correct location in the new browser. This hugely reduces scrolling, but requires some inconvenient window management.

It would be nice to have a "Tab Organizer" page, which presents a Midnight Commander-like side-by-side view of the current window's tab list. This would allow scrolling them independently, and dragging tabs to new locations without causing them to become loaded.

A further extension might be to allow selecting multiple tabs at once, such that dragging the batch would place them consecutively at the drop point.

**Comment by Lej77** [comment]
You can install the extension [Multiple Tab Handler by Piro (piro_or)](https://addons.mozilla.org/firefox/addon/multiple-tab-handler/). After it is installed you can select multiple tabs in the TST sidebar with `ctrl+click`, `shift+click` or `long-press+drag`.

**Comment by eternaleye** [comment]
@Lej77 I tried it, and it really doesn't fit my use case all that well. I want to move a scattered set of tabs from one place in a window to another place in the same window, but that extension seems to give me the following options:

1. Move to new window.
    - This causes unloaded tabs to become loaded; I do not want this.
    - It also doesn't help me move them _back_ to the original window at the location I want.
2. Select multiple, drag.
    - This mostly works, however it's quite sensitive to variation - the boundary between "long enough to pick up and drag" and "so long that it loses my selection and starts a new one" is narrow.
    - In addition, if the tabs to move are far from the desired location, then I still need to rely on the autoscroll-while-dragging.
    - Also, it's finicky when _selecting_ tabs - if my mouse moves too much while control-clicking to add tabs to the selection (and I'm on a touchpad; this happens pretty often) then the selection is lost and the tab this occurred with is dragged alone.
    - In an additional bug, the tab in question got duplicated.

Meanwhile, a Midnight Commander-like page would let me drag tabs one-by-one from current to desired locations, without any risk of losing selection halfway through gathering up the tabs. It would also completely avoid the need to rely on the autoscroll-while-dragging behavior, and could allow for workflows that Multiple Tab Handler _cannot_ support, such as an "insertion sort" workflow (for each tab out of place, put it in the correct place) rather than the "selection sort" workflow (for each place, find every tab that belongs there) that Multiple Tab Handler mandates.

In short: Multiple Tab Handler is a poor fit for what I need.

**Comment by Lej77** [comment]
You can achieve something similar to what you want by opening TST's sidebar page in a separate tab and then moving it to a new window. To open the TST Sidebar page in its own tab see my comments in [#1725](https://github.com/piroor/treestyletab/issues/1725#issuecomment-359856516) and [#1715](https://github.com/piroor/treestyletab/issues/1715#issuecomment-357543989). After you have opened the sidebar page in its own tab you can move the tab to a new window by right clicking it in the sidebar and selecting the `Move to New Window` alternative. When that is done you can drag tabs between the tab list in the new window (not its sidebar) and the current window's sidebar's tab list. Since the tab lists can be scrolled to different positions you can easily insert tabs from the end of the list to the beginning or something like that.

**Comment by eternaleye** [comment]
Yes, but that will still result in loading unloaded tabs, and still requires fiddling with multiple windows so that neither covers the other.

**Comment by lbmaian** [comment]
The "moving tabs loads unloaded tabs" problem is bug #1803.

**Comment by irvinm** [comment]
@eternaleye is there more we need to discuss on this topic?  If no, can you close this item?

**Comment by eternaleye** [comment]
@irvinm,

None of the suggestions yet brought up resolve my use case, of having an interface through which I can organize tabs without juggling multiple windows - even aside from the "moving tabs loads unloaded tabs" issue. In addition, the suggested approaches _also_ suffer from that, by using the actual sidebar code for this, a number of other behaviors very detrimental to that use case come along for the ride (such as the fact that it will, under various circumstances, automatically scroll to the currently focused tab).

**Comment by irvinm** [comment]
@eternaleye ... fair enough. :)

@piroor what are your thoughts on this one?

**Comment by piroor** [comment]
This feature looks too much for me to include as a builtin feature of TST. This addon is designed for a usecase based on automatically constructed tree from links. In other words, usecases with too frequently manual tab organizing is out of purpose.

All suggested features look implementable as a separate addon via TST's API, so I recommend you to try it.
